provider/ecs.go

```diff
@@ -206,13 +206,30 @@ func (provider *ECS) listInstances(ctx context.Context, client *awsClient) ([]ec
 		taskArns = append(taskArns, req.Data.(*ecs.ListTasksOutput).TaskArns...)
 	}
 
-	req, taskResp := client.ecs.DescribeTasksRequest(&ecs.DescribeTasksInput{
-		Tasks:   taskArns,
-		Cluster: &provider.Cluster,
-	})
+	var splitTaskReq [][]*string
```

Comment:
+1 on that one; plus, moving the logic into a dedicated function with an informative name ([as suggested previously](https://github.com/containous/traefik/pull/1209#discussion_r103832525) :-) )